en/docs/visual-programming/data-mapper.md

```diff
@@ -1 +1,160 @@
+
+```ballerina
+function name() => ();
+```
+Once you add the above function, the VS Code plugin will display a code lens called `Design` on top of the function. Click the Design code lens to go to the Data Mapper view.
+
+![Open Data Mapper](img/dm-open-via-code-lens.gif "Open Data Mapper via code lens")
+
+2. Open with the diagram view
+
+Open the file in diagram view and click the `+` icon on the page. Then click Data Mapper in the Add Constructs pane.
+
+![Open Data Mapper](img/dm-open-via-diagram.gif "Open Data Mapper via diagram")
+
+## Provide inputs and output
+Once the Data mapper is opened, it will prompt you to provide inputs and output of the transformation function. Inputs and the output can be any data type in Ballerina. This example converts JSON and array of JSON to JSON, and thereby, you can use Ballerina record types as inputs and output.
+
+In the Data Mapper form, you have several options to provide input and output records. If the records are already defined in your package, you can select one of those. If you are starting from scratch, you can either create the record from the record editor view or import a JSON to create a matching record.
+
+This example imports JSON files and creates the records as shown below.
+
+![Configure Data Mapper](img/dm-define-inputs-n-output.gif "Define Inputs and Output for Data Mapper")
+
+Once you defined inputs and the output types, click Save to open the mapping view. 
+
+## Do the Mappings
+
+Let's see how to map fields in inputs with fields in the output JSON.
+
+### Basic Mapping
+Map the `person id` to the `student id` as shown below.
+>**Info:** The mapping view will have the **Inputs** on the left hand side of the UI and the **Output** on the right. To map the fields, click on the input field port and then, click the output field port. If the input and output fields are compatible and can be mapped directly, you will see a solid line connecting them.
+
+![Basic Mapping](img/dm-basic-mapping.gif "Basic mapping")
+
+### Diagnose and fix mapping errors
+Use the `toBalString` lang lib function to convert the int to string as shown below.
+
+>**Info:** When you map the input to an output fields, some of them might not be compatible due to type mismatch. In this example, if you map the `person age` to `student` age, it will result in an error for type mismatch since the `input age` type is an integer and `output age` type is string. In this case, the Data Mapper will connect the two fields with a red line and show an alert sign. You can see the error by hovering over the alert sign. In this case, it will show `incompatible types: expected 'string', found 'int'`. To fix the error, hover over the alert sign and click `Fix by editing expression`. Then, the Data Mapper will pop out the expression editor for the specific expression. Now, you can modify the expression to return a string.
```

Comment:
```suggestion
>**Info:** When you map the input fields to output fields, some of them might not be compatible due to type mismatch. In this example, if you map the `person age` to `student` age, it will result in an type mismatch error since the `input age` type is an integer and the `output age` type is a string. In this case, the Data Mapper will connect the two fields with a red line and show an alert sign. You can see the error by hovering over the alert sign. In this case, it will show the `incompatible types: expected 'string', found 'int'` error. To fix the error, hover over the alert sign and click **Fix by editing expression**. Then, the Data Mapper will pop out the expression editor for the specific expression. Now, you can modify the expression to return a string.
```